lib/carousel_slider.dart

```diff
@@ -202,52 +204,52 @@ class _CarouselSliderState extends State<CarouselSlider> with TickerProviderStat
 
   @override
   Widget build(BuildContext context) {
-    return getWrapper(PageView.builder(
-      scrollDirection: widget.scrollDirection,
-      controller: widget.pageController,
-      reverse: widget.reverse,
-      itemCount: widget.enableInfiniteScroll ? null : widget.items.length,
-      onPageChanged: (int index) {
-        int currentPage = _getRealIndex(index, widget.realPage, widget.items.length);
-        if (widget.onPageChanged != null) {
-          widget.onPageChanged(currentPage);
+    return getWrapper(NotificationListener<ScrollNotification>(
+      onNotification: (ScrollNotification notification) {
+        if (notification is ScrollUpdateNotification) {
+          setState(() {
+            _page = widget.pageController.page;
+          });
```

Comment:
Adding a `NotificationListener` only to cover the initial state when our `ViewPager` has no initial page seam a little dangerous.
This will trigger every time the carousel scrolls, wich in its default animation speed is around 20 times.
Every callback will update `_page`and call `setState()`, thus re-rendering the the whole carousel.

The existing 'hack' is far from optimal, but it only happens once.

I found this [Medium post](https://medium.com/flutter-community/flutter-laggy-animations-how-not-to-setstate-f2dd9873b8fc) about something similar.

Please correct me if I'm missing something.